gwcs/wcs.py

```diff
@@ -634,3 +641,216 @@ def fix_inputs(self, fixed):
         new_pipeline.append((step0[0], new_transform))
         new_pipeline.extend(self.pipeline[1:])
         return self.__class__(new_pipeline)
+
+    def to_fits_sip(self, bounding_box, max_error=None, max_rms=None, degree=None,
+                    max_inv_error=None, max_inv_rms=None, verbose=False):
```

Comment:
I would like to suggest a different approach on how to specify errors similar to how [`sigma` is specified in `tweakwcs`](https://github.com/spacetelescope/tweakwcs/blob/572159d005afb716b433b702f5692150be704158/tweakwcs/linearfit.py#L113-L124) and possibly use some standard nomenclature, see i.e. [SAS Descriptions of Statistics of Fit](https://documentation.sas.com/?docsetId=fsug&docsetTarget=n1sm8nk3229ttun187529xtkbtpu.htm&docsetVersion=15.1&locale=en).

For example, suggested parameter: `max_fit_error` would replace both `max_error` and `max_rms`:

- `None` (Default)
- `float` (i.e., just a number `2.0`) - 'maxerr' type by default
- `tuple`: `(2.0, 'maxerr')` or `(2.0, 'mae')`, or `(2.0, 'rmse')`
- a `list` of tuples to indicate that _ALL_ conditions must be met: `[(5.0, 'maxerr'), (2.0, 'rmse')]`

I think this is both more clear and more flexible than using the two parameters: `max_error`, `max_rms` because it is more obvious what is going on and also because it allows easy expansions to support other statistics in the future.

Even if you do not accept this suggestion, you should document (maybe in a `Notes` _section_) what happens (how the logic works) when both are `None` and when _both_ are defined. Without looking at the code for a user would be impossible to figure out what all these parameters do and how.

Reply:
This is an interesting possibility, but I think I'd like more comments on whether this is preferred by others. It certainly is easier to generalize, and requires fewer arguments, but on the other hand, needs more explanation so I'm not so sure it is as easy to use.

Reply:
Actually, the more I think about it, the more I lean towards getting rid of `max_rms` altogether. Why would anyone care about RMS when fitting to a know function. I think the most relevant error is `maxerr`: it tells me what will the maximum error be. The RMS could be much smaller due to good fitting, i.e., near CRPIX but one could get really bad errors at the edges and those could be masked by the large number of small errors in most of the image. I think RMS is not informative for this particular problem.

Reply:
It certainly would simplify the code. I tend to agree with you and it is what I would care about (max rather than rms). I suppose we can just support that and see if anyone ever requests anything else.

Reply:
I am also in favor of having only `maxerr` as a start